src/main/java/wolox/training/models/Book.java

```diff
@@ -68,7 +68,81 @@
     @ApiModelProperty(required = true, notes = "Isbn of the book", example = "8993-3232-7628-X")
     private String isbn;
 
+    @Setter
     @ManyToMany(mappedBy = "books")
     private List<User> users = Collections.emptyList();
 
+    public void setGenre(String genre) {
+
+        checkNotNull(genre, "Please check genre field, its null");
+
+        this.genre = genre;
+
+    }
+
+    public void setAuthor(String author) {
+
+        checkNotNull(author, "Please check author field, its null");
+
+        this.author = author;
+
+    }
+
+    public void setImage(String image) {
+
+        checkNotNull(image, "Please check image field, its null");
+
+        this.image = image;
+
+    }
+
+    public void setTitle(String title) {
+
+        checkNotNull(title, "Please check title field, its null");
+
+        this.title = title;
+
+    }
+
+    public void setSubtitle(String subtitle) {
+
+        checkNotNull(subtitle, "Please check subtitle field, its null");
+
+        this.subtitle = subtitle;
+
+    }
+
+    public void setPublisher(String publisher) {
+
+        checkNotNull(publisher, "Please check publisher field, its null");
+
+        this.publisher = publisher;
+
+    }
+
+    public void setYear(String year) {
+
+        checkNotNull(year, "Please check year field, its null");
+
+        this.year = year;
+
+    }
+
+    public void setPages(int pages) {
+
+        checkNotNull(pages, "Please check pages field, its null");
+        checkArgument(pages > 20, "Please check pages field, its can't be 20 or less than 20");
```

Comment:
A good practice is to set constants that represent these seemingly random values, for example `MINIMUM_AMOUNT_OF_PAGES = 20`, this way you improve readability 😄

Reply:
ok, what is called magic numbers.